<!-- -->
<launch>
  <!-- CARLA connection -->
  <arg name='host' default='localhost'/>
  <arg name='port' default='2000'/>
  <arg name='timeout' default='10'/>

  <!-- Map to load on startup (either a predefined CARLA town (e.g. 'Town01'), or a OpenDRIVE map file) -->
  <arg name="track" default='shanghai_intl_circuit'/>

  <!-- Synchronous mode-->
  <arg name='synchronous_mode' default='True'/>
  <arg name='synchronous_mode_wait_for_vehicle_control_command' default='True'/>
  <arg name='fixed_delta_seconds' default='0.05'/>

  <arg name='N' default='1' />
  <arg name='log' default='/tmp/' />

  <arg name='model_type' default='model_free' />
  <arg name='num_wheels' default='4' />

  <arg name='offscreen' default='false' />
  <arg name='vis2D' default='false' />

  <arg name='scenario' default='True' />

  <include file="$(find carla_ros_bridge)/launch/carla_ros_bridge.launch">
    <arg name='host' value='$(arg host)'/>
    <arg name='port' value='$(arg port)'/>
    <arg name='town' value='$(arg track)'/>
    <arg name='timeout' value='$(arg timeout)'/>
    <arg name='synchronous_mode' value='$(arg synchronous_mode)'/>
    <arg name='synchronous_mode_wait_for_vehicle_control_command' value='$(arg synchronous_mode_wait_for_vehicle_control_command)'/>
    <arg name='fixed_delta_seconds' value='$(arg fixed_delta_seconds)'/>
  </include>

  <node pkg="graic_core" type="run.py" name="graic_core" output="screen">
    <param name="host" value="$(arg host)" />
    <param name="port" value="$(arg port)" />
    <param name="N" value="$(arg N)" />
    <param name="log" value="$(arg log)" />
    <param name="track" value="$(arg track)" />
    <param name="model_type" value="$(arg model_type)" />
    <param name="num_wheels" value="$(arg num_wheels)" />
    <param name="offscreen" value="$(arg offscreen)" />
    <param name="scenario" value="$(arg scenario)" />
    <param name="vis2D" value="$(arg vis2D)" />
  </node>

</launch>
